wps-cloudformation-template.yaml

```diff
@@ -57,7 +57,44 @@ Parameters:
   geoserverWpsEndpoint:
     Type: String
     Default: 'http://geoserver-wps.aodn.org.au/geoserver/wps'
+  environment:
+    Type: String
+    Default: dev
+    AllowedValues:
+      - production
+      - systest
+      - sandbox
```

Comment:
Sorry for late review... I don't see the value in putting every single "environment" in here, or referring to environments everywhere through the template.

* fixed, named environments with known bucket names (which don't get deleted)
* everything else

I think it could be more cleanly driven from something like:

```yaml
Default: default
AllowedValues:
    - production
    - systest
    - default
```

This would get rid of a ton of the logic below.

Reply:
I thought it would make sense to include sandbox as it is parallel systest environment. Therefore, the bucket for sandbox should also not get deleted.

Reply:
```yaml
Default: default
AllowedValues:
    - production
    - systest
    - sandbox
    - default
```

Ok so if an environment is promoted to a "named environment", it would just need the following added:

* add name to "environment" list
* add bucket name to "EnvironmentMap" map

Otherwise, it is considered 'default' and ephemeral.

In which case, maybe rename it from "environment" to "EnvironmentConfig", so it's clear that this parameter determines which config is used rather than which environment it actually is.